docs/configuration/listeners/lds.rst

```diff
@@ -0,0 +1,77 @@
+.. _config_listeners_lds:
+
+Listener discovery service
+==========================
+
+The listener discovery service (LDS) is an optional API that Envoy will call to dynamically fetch
+listeners. Envoy will reconcile the API response and add, modify, or remove known listeners
+depending on what is required.
+
+The semantics of listener updates are as follows:
+
+* Every listener must have a unique :ref:`name <config_listeners_name>`. If a name is not
+  provided, Envoy will create a UUID. Listeners that are to be dynamically updated should have a
+  unique name supplied by the management server.
+* When a listener is added, it will be "warmed" before taking traffic. For example, if the listener
+  references an :ref:`RDS <config_http_conn_man_rds>` configuration, that configuration will be
+  resolved and fetched before the listener is moved to "active."
```

Comment:
We probably should take this discussion back to envoy-api or offline, but there are some folks who are wondering why we can't have all the APIs wait until their dependencies are resolved fully before applying. I.e. RDS must also waits for CDS and SDS in this scenario. That LDS has this behavior seems to imply that we're part way there already.